wandb/sdk/artifacts/artifact.py

```diff
@@ -1627,16 +1627,19 @@ def download(
     ) -> FilePathStr:
         """Download the contents of the artifact to the specified root directory.
 
-        Existing files located within `root` are not modified. Explicitly delete
-        `root` before you call `download` if you want the contents of `root` to exactly
-        match the artifact.
+        Existing files located within `root` are not modified. Explicitly delete `root`
+        before you call `download` if you want the contents of `root` to exactly match
+        the artifact.
 
         Arguments:
             root: The directory W&B stores the artifact's files.
             allow_missing_references: If set to `True`, any invalid reference paths
                 will be ignored while downloading referenced files.
-            skip_cache: If set to `True`, the artifact cache will be skipped when downloading
-                and W&B will download each file into the default root or specified download directory.
+            skip_cache: If set to `True`, the artifact cache will be skipped when
+                downloading and W&B will download each file into the default root or
+                specified download directory.
+            path_prefix: If specified, only files with a path that starts with the given
+                prefix will be downloaded. Uses unix format (forward slashes).
```

Comment:
👍

Reply:
follow-up question: As long as the behavior's being changed here, do we want to normalize paths?  E.g. 
- `path/to//prefix` -> `path/to/prefix` (maybe reasonable)
- `path/to/./prefix` -> `path/to/prefix` (ok, maybe still reasonable)
- `path/to/oops/../prefix` -> `path/to/prefix` (on the fence now)

I'm perfectly OK with "no" since it'd minimize the number of changes being made here.

Either way, though, it may be worth clarifying this in the docstring, so users are clearer on what to / what not to expect

Reply:
I'm going to say no for now. Not because of the extra work, but because normalization can lead to unexpected results--like your third example, because it's possible that on the file system in question `path/to/oops/../prefix` actually points somewhere different than `path/to/prefix` due to symlinks.